safe/report/extractors/analysis_provenance_details.py

```diff
@@ -15,6 +20,209 @@
 def analysis_provenance_details_extractor(impact_report, component_metadata):
     """Extracting provenance details of layers.
 
+    This extractor would be the main provenance details extractor which produce
+    tree view provenance details.
+
+    :param impact_report: the impact report that acts as a proxy to fetch
+        all the data that extractor needed
+    :type impact_report: safe.report.impact_report.ImpactReport
+
+    :param component_metadata: the component metadata. Used to obtain
+        information about the component we want to render
+    :type component_metadata: safe.report.report_metadata.
+        ReportComponentsMetadata
+
+    :return: context for rendering phase
+    :rtype: dict
+
+    .. versionadded:: 4.1
+    """
+    context = {}
+    extra_args = component_metadata.extra_args
+
+    default_source = resolve_from_dictionary(
```

Comment:
__[flake8]__
* [F841] local variable 'default_source' is assigned to but never used

<!-- Comment by [SideCI](https://sideci.com) -->